+nosnoc/+discrete_time_problem/Gcs.m

```diff
@@ -0,0 +1,870 @@
+                        ubh = (1+opts.gamma_h)*h0*opts.s_sot_max;
+                        lbh = (1-opts.gamma_h)*h0/opts.s_sot_min;
+                    end
+                    obj.w.h(ii,1:opts.N_finite_elements(ii)) = {{'h', 1}, lbh, ubh, h0};
+                end
+                if obj.opts.step_equilibration == StepEquilibrationMode.linear_complementarity
+                    obj.w.B_max(ii,2:opts.N_finite_elements(ii)) = {{'B_max', dims.n_lambda},-inf,inf};
+                    obj.w.pi_c(ii,2:opts.N_finite_elements(ii)) = {{'pi_c', dims.n_c},-inf,inf};
+                    obj.w.pi_lambda(ii,2:opts.N_finite_elements(ii)) = {{'pi_lambda', dims.n_lambda},-inf,inf};
+                    obj.w.lambda_c(ii,2:opts.N_finite_elements(ii)) = {{'lambda_c', dims.n_c},0,inf};
+                    obj.w.lambda_lambda(ii,2:opts.N_finite_elements(ii)) = {{'lambda_lambda', dims.n_lambda},0,inf};
+                    obj.w.eta(ii,2:opts.N_finite_elements(ii)) = {{'eta', dims.n_lambda},0,inf};
+                    obj.w.nu(ii,2:opts.N_finite_elements(ii)) = {{'nu', 1},0,inf};
+                end
+                if ~opts.right_boundary_point_explicit
+                    obj.w.not_c_primals(ii,1:opts.N_finite_elements(ii)) = {{'not_c_primal', 2}, 0 inf, 0.5};
+                    obj.w.mu_not_c(ii,1:opts.N_finite_elements(ii)) = {{'mu_lambda', 3}, 0 inf, 0.5};
+                end
+            end
+
+            % For c_n ~= 1 case
+            rbp = ~opts.right_boundary_point_explicit;
+            
+            % 3d vars: Variables defined on each rk stage
+            %          some of which are also defined at the initial point:
+            obj.w.x(0,0,opts.n_s) = {{['x_0'], dims.n_x}, model.x0, model.x0, model.x0};
+            obj.w.z(0,0,opts.n_s) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
+            obj.w.lambda(0,0,opts.n_s) = {{['lambda'], dims.n_lambda},0,inf};
+            for ii=1:opts.N_stages
+                if (opts.rk_representation == RKRepresentation.integral ||...
+                    opts.rk_representation == RKRepresentation.differential_lift_x)
+                    obj.w.x(ii,1:opts.N_finite_elements(ii),1:(opts.n_s+rbp)) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+                else
+                    obj.w.x(ii,1:opts.N_finite_elements(ii),opts.n_s) = {{'x', dims.n_x}, model.lbx, model.ubx, model.x0};
+                end
+                if (opts.rk_representation == RKRepresentation.differential ||...
+                    opts.rk_representation == RKRepresentation.differential_lift_x)
+                    obj.w.v(ii,1:opts.N_finite_elements(ii),1:opts.n_s) = {{'v', dims.n_x}};
+                end
+                
+                obj.w.z(ii,1:opts.N_finite_elements(ii),1:(opts.n_s+rbp)) = {{'z', dims.n_z}, model.lbz, model.ubz, model.z0};
+                obj.w.lambda(ii,1:opts.N_finite_elements(ii),1:(opts.n_s+rbp)) = {{'lambda', dims.n_lambda},0, inf};
```

Comment:
what is the default value of lambda? should it be defined in the dcs.Gcs? I guess having an initialization for the complementarity variables not at (0,0) is helpful like in the othe cases.

Reply:
or it should be defined in options. a name for the option would be something like lambda_pds_0 or lambda0_pds

Reply:
I actually don't love having the initial values in the options, and I think having a robust set method is better than per-defining these things. As for not initializing at (0,0) I have actually observed the opposite for some of the PDS problems which is why the current initialization is what it is.

Reply:
For now, I am agnostic about where we put the default lies aslong it is consistent.  
I just dont like them being hard-coded as our wisdom regarding this stuff is quite dynamic. Of course, whatever value leads to good perfomance is fine.
They are not hard coded in the Stewart and Heavside, so I suggest making it consistent.